### PR TITLE
refactor: move retention query class from /ee to /posthog

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- split retention query class between /ee and /posthog

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by neilkakkar — COMMENTED** [review]
`posthog/queries/test/test_retention.py` doesn't seem to test the non-enterprise version of retention. Might be worth bringing this back, and testing with the non-enterprise query?

File: ee/clickhouse/queries/retention/retention_event_query.py [hunk 1]
```diff
@@ -1,245 +1,23 @@
...
-
-    def _determine_should_join_distinct_ids(self) -> None:
-        if (
-            self._filter.aggregation_group_type_index is not None or self._aggregate_users_by_distinct_id
-        ) and not self._column_optimizer.is_using_cohort_propertes:
-            self._should_join_distinct_ids = False
-        else:
-            self._should_join_distinct_ids = True
 
+class ClickhouseRetentionEventsQuery(RetentionEventsQuery, EnterpriseEventQuery):
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply ee/clickhouse/queries/retention/retention_event_query.py]: I changed the import scheme so it should be much more obvious and dynamic now

**Comment by EDsCODE** [comment]
Both comments addressed!

**Review by neilkakkar — APPROVED** [review]
I assume the deleted tests for foss retention pertained to ee functionality?

**Comment by EDsCODE** [comment]
What deleted tests?

**Comment by EDsCODE** [comment]
Oh, hmm let me do one more change

**Pull request merged**